README.md

```diff
@@ -43,34 +43,35 @@ Evidence: Citations
+A commonsense definition of software is that it is "a set of instructions that direct a computer to do a specific task"<sup id="chun2004-footnote">[Chun 2004](#chun2004)</sup>. On the other hand, data is simply a collection of facts or measurements (real or simulated). In other words, software is functionally active, while data is passive. Of course, software (in form) can be considered data as well, especially to functional programmers familiar with LISP and other languages with homoiconicity<sup id="homoiconicity-footnote">[Wikipedia-b](#homoiconicity)</sup>. However, from the point of view of conducting research with software, the main difference is that software is associated with action: knowledge creation, information transformation, visualization, etc.  An action can be thought of a functional transformation between two states of data: a "before" (e.g., input files, parameter settings, unstructured or tacit information) to an "after" state (e.g., output files, transformed data, structured knowledge). That is, software generally performs a function upon something (e.g. software processes data), while data generally has a function performed upon it (e.g. data is processed by software). If we accept the definitions of software and data given at the beginning of this section, then (at least in scientific research), the difference between data and software can be summarized by the statement of <sup id="matthews2010-footnote">[Matthews et al. 2010](#matthews2010)</sup>: "we are more interested in what software *does* rather than what software *is*."
 
+###Software is available in a human readable and separate computer readable form, data in only one of the two.
+
+Especially software in traditional compiled programming languages consist in two forms: the *source code* and the *compiled code*. The transformation between the human readable source code and the machine readable compiled code is performed by a compiler. This is an irreversible process. The existence of both source and compiled code must be taken into account in the license, this makes data collection licenses unsuitable for software and software licenses unsuitable for data collections <sup id="datacommonsfaq-footnote">[DataCommons-faq](#datacommonsfaq)</sup>.
+Furthermore subtle differences between different compilers (e.g. on two different machines or two different operating systems) can result in subtly different results when the software is run on the same data. <sup id="floatcalc-footnote">[Float Differences](#floatcalc)</sup> This puts strong demands on the description of provenance. <sup id="swprov-footnote">[Software Running Provenance](#swprov)</sup>
```

Comment:
I completely agree with the points you make here, but again I don't see what the difference between software and data is.  This says reproducibility is a hard problem, and that running the same software on the same data on a different system/environment can lead to a different result.  How does this show a difference between software and data?